core/src/main/scala-2/sttp/tapir/macros/SchemaMacros.scala

```diff
@@ -39,6 +42,25 @@ trait SchemaCompanionMacros extends SchemaMagnoliaDerivation {
     * [[CreateDerivedEnumerationSchema]].
     */
   def derivedEnumeration[T]: CreateDerivedEnumerationSchema[T] = macro SchemaEnumerationMacro.derivedEnumeration[T]
+
+  /** Create a schema for scala `Enumeration` and the `Validator` instance based on possible enumeration values */
+  implicit def schemaForEnumeration[T <: scala.Enumeration#Value](implicit
+      tag: TypeTag[T],
+      annotations: SchemaAnnotations[T]
+  ): Schema[T] = {
```

Comment:
This is based on [avro4s](https://github.com/sksamuel/avro4s/blob/f41e25c934d6f475eca6127daedfd8183d1afa2f/avro4s-core/src/main/scala/com/sksamuel/avro4s/BaseTypes.scala#:~:text=implicit%20def-,scalaEnumSchemaFor,-%5BE%20%3C%3A%20scala.Enumeration). I tried with macro but it seemed not possible. This is only for scala2 since in scala3 we have no `TypeTag` (so it should be doable with macro).